new PR, changes while absent

JohnBurdick1 opened the pull request:
-Added tickToAngle method in arm part classes/Modified getAngle method to account for this.
-Removed needless imports + uncalled CANCoder config method.
-Slight comment changes.
-Please message me on Slack if other changes must be made. Thank you for understanding my absence.

JohnBurdick1 committed:
Commit, because I will be absent today

-Added tickToAngle method in arm part classes
-Removed needless imports + uncalled CANCoder config method
-Slight comment changes
-Please message me on Slack if other changes need to be made.

MathNerd28 commented on src/main/java/frc/robot/subsystems/arm/ArmBase.java:
```diff
@@ -45,20 +39,9 @@ public ArmBase() {
         feedforward = new ArmFeedforward(kS_BASE, kG_BASE, kV_BASE, kA_BASE);
     }
-    
-    /**private void configCancoder() {
-        CANCoderConfiguration config = new CANCoderConfiguration();
-        config.absoluteSensorRange = AbsoluteSensorRange.Unsigned_0_to_360;
-        config.magnetOffsetDegrees = 0;
-        config.sensorDirection = false;
-        config.initializationStrategy = SensorInitializationStrategy.BootToAbsolutePosition;
-        canCoder.configAllSettings(config);
-        canCoder.setPosition(
-                canCoder.getAbsolutePosition() - BASE_CC_OFFSET);
-    }*/
```
You shouldn’t delete this method, as it will likely prove useful once the CANCoder is added again. For now, just uncomment it. Since you aren’t referencing it anywhere else, it will never be executed, so you don’t have to worry about the device not existing. (This is called “dead code”)

MathNerd28 commented on src/main/java/frc/robot/subsystems/arm/ArmElbow.java:
```diff
@@ -44,17 +39,6 @@ public ArmElbow() {
         feedforward = new ArmFeedforward(kS_ELBOW, kG_ELBOW, kV_ELBOW, kA_ELBOW);
     }
 
-    /**private void configCancoder() {
-        CANCoderConfiguration config = new CANCoderConfiguration();
-        config.absoluteSensorRange = AbsoluteSensorRange.Unsigned_0_to_360;
-        config.magnetOffsetDegrees = 0;
-        config.sensorDirection = false;
-        config.initializationStrategy = SensorInitializationStrategy.BootToAbsolutePosition;
-        canCoder.configAllSettings(config);
-        canCoder.setPosition(
-                canCoder.getAbsolutePosition() - ELBOW_CC_OFFSET);
-    }*/
```
Same comment as `ArmBase.java`

MathNerd28 commented on src/main/java/frc/robot/subsystems/arm/ArmWrist.java:
```diff
@@ -42,17 +37,6 @@ public ArmWrist() {
         feedforward = new ArmFeedforward(kS_ELBOW, kG_ELBOW, kV_ELBOW, kA_ELBOW);
     }
-
-    /**private void configCancoder() {
-        CANCoderConfiguration config = new CANCoderConfiguration();
-        config.absoluteSensorRange = AbsoluteSensorRange.Unsigned_0_to_360;
-        config.magnetOffsetDegrees = 0;
-        config.sensorDirection = false;
-        config.initializationStrategy = SensorInitializationStrategy.BootToAbsolutePosition;
-        canCoder.configAllSettings(config);
-        canCoder.setPosition(
-                canCoder.getAbsolutePosition() - ARM_WRIST_CC_OFFSET);
-    }*/
```
Same comment as `ArmBase.java`

MathNerd28 commented:
Also, you should merge latest master into your branch before merging back

mmarchetti commented:
@MathNerd28 Since John is out, can you make the proposed changes on this branch? We need the `getAngle` change before we can test closed-loop control tonight.

DevilTech committed:
started arm tuning

Currently oscillates because we introduced kI. We are using iZone set to 5 degrees to prevent excessive integral response. 2 degress was not enough. We should try values in between; if still not right we can adjust ki.

mmarchetti committed:
factor out a common ArmSegment base class

JohnBurdick1 committed:
changes before pulling code

JohnBurdick1 committed:
Merge branch 'arm' of https://github.com/Team1559/2023-Charged-Up into arm

JohnBurdick1 committed:
Merge remote-tracking branch 'origin/arm-base-class' into arm

JohnBurdick1 committed:
Tuned PID + FeedForward constants.

JohnBurdick1 committed:
final commit 2/11/23

-Tuned PID + FF constants to account for a fully extended, worst-case-scenario arm
-changed some array positions
-changed selectedSensorPosition sets to 60
-changed some comments

JohnBurdick1 committed:
Commit for 2/14/23

JohnBurdick1 committed:
Mid-Meeting Commit 2/15/23

JohnBurdick1 committed:
Final Commit 2/15/23

JohnBurdick1 committed:
commit for 2/16/23

MathNerd28 committed:
Merge branch 'master' into arm

MathNerd28 committed:
Rename main arm file + implement dynamic kG

- Rename 'FullArmCommands' to 'Arm' for simplicity
- Add double linked list references to 'ArmSegment'
- Implement 'ArmSegment' dynamic kG calculation based on modular system center of mass

MathNerd28 committed:
Remove unnecesssary logging, commit before removing some math due to mechanical assembly

MathNerd28 committed:
Multi-axis control!

MathNerd28 committed:
Merge branch 'master' into arm

MathNerd28 committed:
Format ArmSegment.java

MathNerd28 committed:
Added smooth arm acceleration and deceleration

MathNerd28 committed:
Add kV, kA (elbow somewhat stable)

MathNerd28 committed:
Fix pull request changes
